src/jade/tabs/contacts.jade

```diff
@@ -99,7 +99,7 @@ section.col-xs-12.content(ng-controller="ContactsCtrl")
       span.name(ng-hide='editing', ng-bind="entry.name") 
     .col-xs-12.col-sm-6.col-md-6(ng-hide='editing')
       div
-        b.hidden-xs {{entry.address | rpaddressorigin}}: {{entry.address}}
+        b.hidden-xs {{entry.address | rpaddressorigin}}: {{contactNames[entry.address] || resolveName(entry.address)}}
```

Comment:
Initially, nothing is in `contactNames`, so there should be many cases when `entry.address` is resolvable but not yet in `contactNames`. `resolveName` returns `entry.address` so that until it _does_ resolve, the address is shown.